src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -34,8 +34,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             if ($options['multiple']) {
                 $data = array();
+                $files = $event->getData();
 
-                foreach ($event->getData() as $file) {
+                if (!is_array($files)) {
+                    $files = array(null);
```

Comment:
`$files = array();` also looks more better to me, it reduces the redundant traverse over them and `RequestHandlerInterface::isFileUpload()` method call.

Reply:
fixed while merging